BoxContentSDK/BoxContentSDK/Operations/BOXAPIDataOperation.m

```diff
@@ -395,9 +367,14 @@ - (id)copyWithZone:(NSZone *)zone
     return operationCopy;
 }
 
-- (BOOL)canBeReenqueued
+- (BOOL)canBeReenqueuedDueToTokenExpired
 {
     return self.shouldStartImmediately == NO;
 }
 
+- (BOOL)canBeReenqueuedDueTo202NotReady
+{
+    return YES;
+}
+
```

Comment:
Notice the logic/semantics here (pre-existing logic). BOXAPIDataOperation recently started answering the question "Should you be re-enqueued after a token expired error?" with the response "Yes, if shouldStartImmediately ==NO". 

But at the *Same* time BOXAPIDataOperation would always re-enqueue after a 202 response.

The `self.shouldStartImmediately == NO` was recently added (PR 648). I'm concerned that this is an issue. That PR 648 might have missed the logic around 202 handling. This applies to BoxStreamOperation and other 202 handling Operation classes.

Reply:
The reason that re-enqueueing is disabled when `shouldStartImmediately` is that we're asking to skip the queue entirely in that case, so it would be very odd to have the request enqueued anyway.

Reply:
Yes, that's my point. Before Thuy's change this code would re-enqueue logic wasn't disabled at all. After Thuy's change the TokenExpired logic "respected" the `startImmediately` value, but the 202 logic does not.

If we're in agreement over this I can tweak this PR to address that.

Reply:
The ultimate reason to want to avoid the queue is that using the queue introduces a race condition in the operations code. I believe that, in order to avoid effectively the same race in that code, we have to let the operations queue manage retry.